### ISSUE TITLE
Defining other methods in an ActiveRecord model doesn't work

### ISSUE THREAD

**Issue by cimm** [jeffp/enumerated_attribute#3]
I'm trying to define my own methods like the 'parked?' and 'driving?' examples but can't get it to work. It works as long as the attribute is not a column in my database.

```
class Tractor < ActiveRecord::Base
  enum_attr :gear, %w(reverse ^neutral first second over_drive) do
    driving? [:first, :second, :over_drive]
  end
end

t = Tractor.new
t.driving?                        # => false
```

Works when 'gear' is not a column in the Tractors table but does not work when 'gear' is defined as a t.enum field in the migration (it will always return false in that case). It seems this has something to do with the string versus symbol comparison.

Using MySQL 5.1.40 and Ruby 1.8.7 with Rails 2.3.5.

**Comment by petyosi** [comment]
Confirming that problem too.

**Comment by petyosi** [comment]
Should be fixed with http://github.com/underlog/enumerated_attribute/commit/f45543073b654f100293664fc7bba3ab4601d0ee

**Comment by cimm** [comment]
Cool, thanks for the fix! Lets hope Jeff can pull your fix.

**Comment by turadg** [comment]
Looks good. @underlog, if you submit a pull request I'll merge it.

**Comment by petyosi** [comment]
One year later :). Did one.

**Comment by turadg** [comment]
@underlog, great thanks. I merged it but before pushing this out in a new gem, we should have add a test that failed before and passes now. @cimm?

**Comment by cimm** [comment]
@turadg Indeed, I should have added a test. No longer maintain the app I used it for so I can't immediately verify it, sorry.
